src/test/java/ConnectBoardTest.java

```diff
@@ -0,0 +1,42 @@
+package remer.connect4;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ConnectBoardTest
+{
+    @Test
+    public void insert()
+    {
+        ConnectBoard board = new ConnectBoard(7, 6);
+        board.insert(3, 'X');
+
+        assertEquals('X', board.toString().split("\n")[5].charAt(3));
```

Comment:
If you look at the stack trace, it will show you **where** this error occurs.
With that, and the error message, try to figure out what your bug is.